app/terminal/terminal.tsx

```diff
@@ -72,6 +75,27 @@ export default class TerminalComponent extends React.Component<TerminalProps, Te
   }
 
   handleDownloadClicked() {
+    if (typeof this.props.invocationId === "string") {
```

Comment:
nit: equivalently, you can write `if (this.props.invocationId)`

Reply:
Currently, we (mostly) treat empty / zero values, undefined, and null the same, so `typeof` is rarely needed.

Eventually I think it'd be nice to decide on a single value to represent a non-present optional value (either `undefined` or `null`) and enable strict null checks in our `tsconfig.json`. Then, when trying to decide whether a value is present, we can just check `!== null` (or `!== undefined`).